### DISCUSSION TITLE
Prevent special character in variable from executing on echo

### DISCUSSION THREAD

**Discussion by dakshshah96** [community/community#25947] [Actions]
<p>I have a GitHub Actions job where I check whether to continue with deployment by checking the contents of the commit message. Deployment happens only when the commit message is exactly this: <code>chore(release): publish</code>.</p>
<pre><code class="lang-auto">name: Test &amp; Deploy Packages
on:
  push:
    branches:
      - master

jobs:
  check_if_deploy:
    runs-on: ubuntu-latest
    outputs:
      shouldContinue: ${{ steps.decideContinue.outputs.shouldContinue }}

    steps:
    - name: Checkout source code
      uses: actions/checkout@v2

    - name: Get GitHub context
      env:
        GITHUB_CONTEXT: ${{ toJson(github) }}
      run: echo "$GITHUB_CONTEXT"

    - name: Decide whether to continue based on commit message
      id: decideContinue
      run: |
        commitMessage=$(git log --oneline --format=%B -n 1 ${{ github.event.after }} | head -n 1)
        echo $commitMessage
        if [[$commitMessage == "chore(release): publish"]]; then
          echo "::set-output name=shouldContinue::true"
          echo "publish commit, continuing with deployment"
        else
          echo "::set-output name=shouldContinue::false"
          echo "not a publish commit, skipping deployment"
        fi
</code></pre>
<p>My next job checks whether <code>needs.check_if_deploy.outputs.shouldContinue</code> is <code>true</code> and continues or skips based on that condition. This seems to be working fine.</p>
<p>However, a <code>command not found</code> error is always printed during execution of the <code>check_if_deploy</code> job. Line 13 in the below screenshot says <code>/home/runner/work/_temp/ba7b09fc-462a-4755-ad83-a52a8c725c55.sh: line 3: [[chore:: command not found</code>:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/f/f5a8336be09381379e5e3aa4f8a9a079956d74af.png" data-download-href="/uploads/short-url/z3blDmLPklGlGM1OTOQ08Pea6yH.png?dl=1" title="Screenshot 2020-07-18 10.24.53" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/7896438/181094265-721022b1-0873-438c-8ba8-66ac258e2edb.png" alt="Screenshot 2020-07-18 10.24.53" data-base62-sha1="z3blDmLPklGlGM1OTOQ08Pea6yH" width="690" height="369" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/f/f5a8336be09381379e5e3aa4f8a9a079956d74af_2_690x369.png, https://aws1.discourse-cdn.com/github/original/2X/f/f5a8336be09381379e5e3aa4f8a9a079956d74af.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/f/f5a8336be09381379e5e3aa4f8a9a079956d74af.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/f5a8336be09381379e5e3aa4f8a9a079956d74af_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-07-18 10.24.53</span><span class="informations">875×469 54.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I understand this has something to do with the <code>echo</code> on line 3 executing the contents of the variable instead of just printing it but don’t know how to resolve it.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The problem is that there are no spaces after <code>[[</code> and before <code>]]</code>. The <code>[[</code> is a bash command, and without the space it’s combined with the first word in the <code>$commitMessage</code> variable instead, which is no valid command. Additionally string comparison with the <code>[[</code> or <code>[</code> commands is just one equals sign. Finally you should quote both sides of the comparison.</p>
<p>So your if needs to be:</p>
<pre><code class="lang-bash">        if [[ "$commitMessage" = "chore(release): publish" ]]; then
</code></pre>

**Comment by dakshshah96** [comment]
<p>Thank you so much! Never realised that this was the problem.</p>
